### PR TITLE
[DOCS] Added docstring for isclose() utility method

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/util.py [hunk 1]
```diff
@@ -1320,12 +1320,27 @@ def is_nan(value: Any) -> bool:
 def isclose(
     operand_a: Union[datetime.datetime, Number],
     operand_b: Union[datetime.datetime, Number],
-    rtol=1.0e-5,
-    atol=1.0e-8,
+    rtol=1.0e-5,  # controls relative weight of "operand_b" (when its magnitude is large)
+    atol=1.0e-8,  # controls absolute accuracy (based on floating point machine precision)
```
Comment (Shinnnyshinshin) [inline great_expectations/util.py:1324]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
Love it. LGTM

File: great_expectations/util.py [hunk 2]
```diff
@@ -1320,12 +1320,27 @@ def is_nan(value: Any) -> bool:
...
+    This translates to:
+        "absolute(operand_a - operand_b) <= (atol + rtol * absolute(operand_b))", where "operand_a" is "target" quantity
+    under evaluation for being close to a "control" value, and "operand_b" serves as the "control" ("reference") value.
+
+    The values of the absolute tolerance ("atol") parameter is chosen as a sufficiently small constant for most floating
+    point machine representations (e.g., 1.0e-8), so that even if the "control" value is small in magnitude and "target"
+    and "control" are close in absolute value, then the accuracy of the assessment can still be high up to the precision
+    of the "atol" value (here, 8 digits as the default).  However, when the "control" value is large in magnitude, the
+    relative tolerance ("rtol") parameter carries a greater weight in the comparison assessment, because the acceptable
+    deviation between the two quantities can be relatively larger for them to be deemed as "close enough" in this case.
```
Comment (NathanFarmer) [inline great_expectations/util.py:1343]: ❤️

**Review by NathanFarmer — COMMENTED** [review]
Thank you for this!

**Pull request merged**